components/support/integration/src/interrupt.rs

```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Helps manage "interruptable" things across our various crates.
+
+use crate::error::*;
+
+/// Something that is interruptable. In practice, this will almost certainly
+/// be Sync + Send, as it will typically be created on one thread, but
+/// `.interrupt()` will be called from a different thread. However,
+/// Sync + Send semantics aren't mandated here.
+pub trait Interruptable {
```

Comment:
This is more general than I think we need (and this generality was why you had trouble with the `'static` bound). That said, for now this is fine, and we can consider simplifying later..